src/unix/bsd/freebsdlike/mod.rs

```diff
@@ -606,8 +563,12 @@ extern {
                            mibp: *mut ::c_int,
                            sizep: *mut ::size_t)
                            -> ::c_int;
+    #[cfg(not(target_os = "dragonfly"))]
     pub fn mprotect(addr: *const ::c_void, len: ::size_t, prot: ::c_int)
                     -> ::c_int;
+    #[cfg(target_os = "dragonfly")]
+    pub fn mprotect(addr: *mut ::c_void, len: ::size_t, prot: ::c_int)
+                    -> ::c_int;
```

Comment:
Could you push this definition and `clock_gettime` below into the respective platform-specific modules?